pkg/standalone-utils/test/LidoRelayer.test.ts

```diff
@@ -1,461 +1,838 @@
+          expectTransferEvent(
+            receipt,
+            {
+              from: TypesConverter.toAddress(relayerIsRecipient ? wstETH : relayer),
+              to: TypesConverter.toAddress(relayerIsRecipient ? relayer : tokenRecipient),
+              value: await wstETH.instance.getStETHByWstETH(amount),
+            },
+            stETH
+          );
+        });
+      }
     });
 
-    context('when the relayer is authorized', () => {
-      sharedBeforeEach('allow relayer', async () => {
-        const manageUserBalanceAction = await actionId(vault.instance, 'manageUserBalance');
-        const batchSwapAction = await actionId(vault.instance, 'batchSwap');
+    describe('stakeETH', () => {
+      let tokenRecipient: Account;
 
-        await vault.authorizer
-          ?.connect(admin)
-          .grantRolesGlobally([manageUserBalanceAction, batchSwapAction], relayer.address);
+      context('recipient = senderUser', () => {
+        beforeEach(() => {
+          tokenRecipient = senderUser;
+        });
+        testStake();
       });
 
-      context('when the user did allow the relayer', () => {
-        sharedBeforeEach('allow relayer', async () => {
-          await vault.instance.connect(sender).setRelayerApproval(sender.address, relayer.address, true);
-        });
-
-        describe('swap using stETH as an input', () => {
-          sharedBeforeEach('build swap request', async () => {
-            swaps = [
-              {
-                poolId: basePoolId,
-                assetInIndex: basePoolTokens.findIndexBySymbol('wstETH'),
-                assetOutIndex: basePoolTokens.findIndexBySymbol('WETH'),
-                amount: fp(1),
-                userData: '0x',
-              },
-            ];
-
-            assets = basePoolTokens.addresses;
-            limits = basePoolTokens.map((token) => (token.symbol === 'wstETH' ? fp(1) : 0));
-          });
+      context('recipient = relayer', () => {
+        beforeEach(() => {
+          tokenRecipient = relayer;
+        });
+        testStake();
+      });
 
-          it('performs the given swap', async () => {
-            const receipt = await relayer
-              .connect(sender)
-              .batchSwap(SwapKind.GivenIn, swaps, assets, funds, limits, deadline);
-
-            expectEvent.inIndirectReceipt(await receipt.wait(), vault.instance.interface, 'Swap', {
-              poolId: swaps[0].poolId,
-              tokenIn: assets[swaps[0].assetInIndex],
-              tokenOut: assets[swaps[0].assetOutIndex],
-              amountIn: swaps[0].amount,
-              // amountOut
-            });
-          });
+      function testStake(): void {
+        it('stakes with immediate amounts', async () => {
+          const receipt = await (
+            await relayer.connect(senderUser).multicall([encodeStakeETH(tokenRecipient, amount)], { value: amount })
+          ).wait();
```

Comment:
Slightly redundant, but it might not hurt to have an extra test here where `value > amount`, and we check that the excess ETH is returned.